{"parents": [{"link": "../", "title": "Release notes"}], "prev": {"link": "../1.5.3/", "title": "Django 1.5.3 release notes"}, "next": {"link": "../1.5.1/", "title": "Django 1.5.1 release notes"}, "title": "Django 1.5.2 release notes", "meta": {}, "body": "<div class=\"section\" id=\"s-django-1-5-2-release-notes\">\n<span id=\"django-1-5-2-release-notes\"></span><h1>Django 1.5.2 release notes<a class=\"headerlink\" href=\"#django-1-5-2-release-notes\" title=\"Permalink to this headline\">\u00b6</a></h1>\n<p><em>August 13, 2013</em></p>\n<p>This is Django 1.5.2, a bugfix and security release for Django 1.5.</p>\n<div class=\"section\" id=\"s-mitigated-possible-xss-attack-via-user-supplied-redirect-urls\">\n<span id=\"mitigated-possible-xss-attack-via-user-supplied-redirect-urls\"></span><h2>Mitigated possible XSS attack via user-supplied redirect URLs<a class=\"headerlink\" href=\"#mitigated-possible-xss-attack-via-user-supplied-redirect-urls\" title=\"Permalink to this headline\">\u00b6</a></h2>\n<p>Django relies on user input in some cases (e.g.\n<code class=\"docutils literal notranslate\"><span class=\"pre\">django.contrib.auth.views.login()</span></code>, <code class=\"docutils literal notranslate\"><span class=\"pre\">django.contrib.comments</span></code>, and\n<a class=\"reference internal\" href=\"../../topics/i18n/\"><span class=\"doc\">i18n</span></a>) to redirect the user to an \u201con success\u201d URL.\nThe security checks for these redirects (namely\n<code class=\"docutils literal notranslate\"><span class=\"pre\">django.utils.http.is_safe_url()</span></code>) didn\u2019t check if the scheme is <code class=\"docutils literal notranslate\"><span class=\"pre\">http(s)</span></code>\nand as such allowed <code class=\"docutils literal notranslate\"><span class=\"pre\">javascript:...</span></code> URLs to be entered. If a developer\nrelied on <code class=\"docutils literal notranslate\"><span class=\"pre\">is_safe_url()</span></code> to provide safe redirect targets and put such a\nURL into a link, they could suffer from a XSS attack. This bug doesn\u2019t affect\nDjango currently, since we only put this URL into the <code class=\"docutils literal notranslate\"><span class=\"pre\">Location</span></code> response\nheader and browsers seem to ignore JavaScript there.</p>\n</div>\n<div class=\"section\" id=\"s-xss-vulnerability-in-django-contrib-admin\">\n<span id=\"xss-vulnerability-in-django-contrib-admin\"></span><h2>XSS vulnerability in <a class=\"reference internal\" href=\"../../ref/contrib/admin/#module-django.contrib.admin\" title=\"django.contrib.admin: Django's admin site.\"><code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">django.contrib.admin</span></code></a><a class=\"headerlink\" href=\"#xss-vulnerability-in-django-contrib-admin\" title=\"Permalink to this headline\">\u00b6</a></h2>\n<p>If a <a class=\"reference internal\" href=\"../../ref/models/fields/#django.db.models.URLField\" title=\"django.db.models.URLField\"><code class=\"xref py py-class docutils literal notranslate\"><span class=\"pre\">URLField</span></code></a> is used in Django 1.5, it displays the\ncurrent value of the field and a link to the target on the admin change page.\nThe display routine of this widget was flawed and allowed for XSS.</p>\n</div>\n<div class=\"section\" id=\"s-bugfixes\">\n<span id=\"bugfixes\"></span><h2>Bugfixes<a class=\"headerlink\" href=\"#bugfixes\" title=\"Permalink to this headline\">\u00b6</a></h2>\n<ul class=\"simple\">\n<li>Fixed a crash with <a class=\"reference internal\" href=\"../../ref/models/querysets/#django.db.models.query.QuerySet.prefetch_related\" title=\"django.db.models.query.QuerySet.prefetch_related\"><code class=\"xref py py-meth docutils literal notranslate\"><span class=\"pre\">prefetch_related()</span></code></a>\n(#19607) as well as some <code class=\"docutils literal notranslate\"><span class=\"pre\">pickle</span></code> regressions with <code class=\"docutils literal notranslate\"><span class=\"pre\">prefetch_related</span></code>\n(#20157 and #20257).</li>\n<li>Fixed a regression in <a class=\"reference internal\" href=\"../../ref/contrib/gis/#module-django.contrib.gis\" title=\"django.contrib.gis: Geographic Information System (GIS) extensions for Django\"><code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">django.contrib.gis</span></code></a> in the Google Map output on\nPython 3 (#20773).</li>\n<li>Made <code class=\"docutils literal notranslate\"><span class=\"pre\">DjangoTestSuiteRunner.setup_databases</span></code> properly handle aliases for\nthe default database (#19940) and prevented <code class=\"docutils literal notranslate\"><span class=\"pre\">teardown_databases</span></code> from\nattempting to tear down aliases (#20681).</li>\n<li>Fixed the <code class=\"docutils literal notranslate\"><span class=\"pre\">django.core.cache.backends.memcached.MemcachedCache</span></code> backend\u2019s\n<code class=\"docutils literal notranslate\"><span class=\"pre\">get_many()</span></code> method on Python 3 (#20722).</li>\n<li>Fixed <a class=\"reference internal\" href=\"../../ref/contrib/humanize/#module-django.contrib.humanize\" title=\"django.contrib.humanize: A set of Django template filters useful for adding a &quot;human touch&quot; to data.\"><code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">django.contrib.humanize</span></code></a> translation syntax errors. Affected\nlanguages: Mexican Spanish, Mongolian, Romanian, Turkish (#20695).</li>\n<li>Added support for wheel packages (#19252).</li>\n<li>The CSRF token now rotates when a user logs in.</li>\n<li>Some Python 3 compatibility fixes including #20212 and #20025.</li>\n<li>Fixed some rare cases where <a class=\"reference internal\" href=\"../../ref/models/querysets/#django.db.models.query.QuerySet.get\" title=\"django.db.models.query.QuerySet.get\"><code class=\"xref py py-meth docutils literal notranslate\"><span class=\"pre\">get()</span></code></a>\nexceptions recursed infinitely (#20278).</li>\n<li><a class=\"reference internal\" href=\"../../ref/django-admin/#django-admin-makemessages\"><code class=\"xref std std-djadmin docutils literal notranslate\"><span class=\"pre\">makemessages</span></code></a> no longer crashes with <code class=\"docutils literal notranslate\"><span class=\"pre\">UnicodeDecodeError</span></code>\n(#20354).</li>\n<li>Fixed <code class=\"docutils literal notranslate\"><span class=\"pre\">geojson</span></code> detection with SpatiaLite.</li>\n<li><a class=\"reference internal\" href=\"../../topics/testing/tools/#django.test.SimpleTestCase.assertContains\" title=\"django.test.SimpleTestCase.assertContains\"><code class=\"xref py py-meth docutils literal notranslate\"><span class=\"pre\">assertContains()</span></code></a> once again works with\nbinary content (#20237).</li>\n<li>Fixed <a class=\"reference internal\" href=\"../../ref/models/fields/#django.db.models.ManyToManyField\" title=\"django.db.models.ManyToManyField\"><code class=\"xref py py-class docutils literal notranslate\"><span class=\"pre\">ManyToManyField</span></code></a> if it has a Unicode <code class=\"docutils literal notranslate\"><span class=\"pre\">name</span></code>\nparameter (#20207).</li>\n<li>Ensured that the WSGI request\u2019s path is correctly based on the\n<code class=\"docutils literal notranslate\"><span class=\"pre\">SCRIPT_NAME</span></code> environment variable or the <a class=\"reference internal\" href=\"../../ref/settings/#std-setting-FORCE_SCRIPT_NAME\"><code class=\"xref std std-setting docutils literal notranslate\"><span class=\"pre\">FORCE_SCRIPT_NAME</span></code></a>\nsetting, regardless of whether or not either has a trailing slash (#20169).</li>\n<li>Fixed an obscure bug with the <a class=\"reference internal\" href=\"../../topics/testing/tools/#django.test.override_settings\" title=\"django.test.override_settings\"><code class=\"xref py py-func docutils literal notranslate\"><span class=\"pre\">override_settings()</span></code></a>\ndecorator. If you hit an <code class=\"docutils literal notranslate\"><span class=\"pre\">AttributeError:</span> <span class=\"pre\">'Settings'</span> <span class=\"pre\">object</span> <span class=\"pre\">has</span> <span class=\"pre\">no</span> <span class=\"pre\">attribute</span>\n<span class=\"pre\">'_original_allowed_hosts'</span></code> exception, it\u2019s probably fixed (#20636).</li>\n</ul>\n</div>\n</div>\n", "metatags": "", "rellinks": [["genindex", "General Index", "I", "index"], ["py-modindex", "Python Module Index", "", "modules"], ["releases/1.5.1", "Django 1.5.1 release notes", "N", "next"], ["releases/1.5.3", "Django 1.5.3 release notes", "P", "previous"]], "sourcename": "releases/1.5.2.txt", "toc": "<ul>\n<li><a class=\"reference internal\" href=\"#\">Django 1.5.2 release notes</a><ul>\n<li><a class=\"reference internal\" href=\"#mitigated-possible-xss-attack-via-user-supplied-redirect-urls\">Mitigated possible XSS attack via user-supplied redirect URLs</a></li>\n<li><a class=\"reference internal\" href=\"#xss-vulnerability-in-django-contrib-admin\">XSS vulnerability in <code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">django.contrib.admin</span></code></a></li>\n<li><a class=\"reference internal\" href=\"#bugfixes\">Bugfixes</a></li>\n</ul>\n</li>\n</ul>\n", "display_toc": true, "page_source_suffix": ".txt", "current_page_name": "releases/1.5.2", "sidebars": ["localtoc.html", "relations.html", "sourcelink.html", "searchbox.html"], "customsidebar": null, "favicon_url": null, "logo_url": null, "alabaster_version": "0.7.13", "include_console_assets": false}